dev/buildtool/source_commands.py

```diff
@@ -138,6 +141,97 @@ def init_argparser(self, parser, defaults):
         )
 
 
+class TagBranchCommand(RepositoryCommandProcessor):
+    """Implements the tag_branch command."""
+
+    def __init__(self, factory, options):
+        """Implements CommandProcessor interface."""
+
+        self.__git = GitRunner(options)
+
+        all_names = list(SPINNAKER_RUNNABLE_REPOSITORY_NAMES)
+        all_names.extend(SPINNAKER_RUNNABLE_NON_CORE_REPOSITORY_NAMES)
+        all_names.extend(SPIN_REPOSITORY_NAMES)
+        super().__init__(factory, options, source_repository_names=all_names)
+
+    def _do_repository(self, repository):
```

Comment:
For each repository in `all_names` above, perform the following:
1. clone repo
2. get latest tag & its commit ID
3. compare to HEAD commit ID, if it matches do nothing for this repo.
4. generate a new minor or patch tag branch depending
5. push tag to git remote